tasks/support/github_pr_stats.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+# custom error class
+class GithubAPIError < StandardError
+  attr_reader :object
+
+  def initialize(object = nil)
+    @object = object
+  end
+end
+
+# fetch open pull requests
+class GithubPRStats
+  attr_reader :token
+
+  def initialize
+    @config_file = '_config.yml'
+    @config = YAML.safe_load(File.read(@config_file))
+    @config['description'].chomp!("\n")
+    api_url = 'https://api.github.com/orgs/voxpupuli/repos?per_page=100'
+    oauth_client_id = ENV['GH_CLIENT_ID'] ||= nil
+    oauth_client_secret = ENV['GH_CLIENT_SECRET'] ||= nil
+    @token = oauth_client_secret
+    @auth_suffix = nil
+    if oauth_client_id && oauth_client_secret
+      @auth_suffix = "&client_id=#{oauth_client_id}&client_secret=#{oauth_client_secret}"
+    end
+    @api_url = api_url + @auth_suffix.to_s
+    @repos = 0
+    @pull_reqs = 0
+  end
+
+  def query(url: @api_url, page: 1)
+    url = URI.parse(url + "&page=#{page}")
+    req = Net::HTTP::Get.new(url.to_s)
+    req['Accept'] = 'application/vnd.github.v3+json'
+    req['User-Agent'] = 'voxpupuli.org github API'
+    res = Net::HTTP.start(url.host, url.port, use_ssl: true) { |http| http.request(req) }
+
+    raise GithubAPIError.new, "API response status: #{res.code}" if res.code.to_s =~ %r{(4|5)\d\d}
+
+    # see if there are more pages
+    next_page = nil
+    if res['link']
+      res['link'].split(',').each do |link|
+        # also catch possibly nested quoting
+        if link =~ %r{rel=(\\|)"next(\\|)"}
+          next_url = URI.extract(link)[0]
+          next_page = next_url.match(%r{(&|\?)page=(?<page>[0-9]+)})[:page]
```

Comment:
yes, but they tell you everywhere to use that HTTP Link header to traverse the result pages...